pkg/bgpv1/gobgp/workdiff.go

```diff
@@ -61,11 +63,41 @@ func (wd *reconcileDiff) diff(m LocalASNMap, policy *v2alpha1api.CiliumBGPPeerin
 
 // String provides a string representation of the reconcileDiff.
 func (wd *reconcileDiff) String() string {
-	return fmt.Sprintf("Registering: %v Withdrawing: %v Reconciling: %v",
-		wd.register,
-		wd.withdraw,
-		wd.reconcile,
-	)
+	var register, withdraw, reconcile strings.Builder
+	// The length of register will be len(wd.register) + 2 + len(wd.register) - 1
+	// e.g wd.register := []int{1,2,3,4}, then the length of register will be
+	// len([]int{1,2,3,4}) + len("[]") + len("[]string{" ", " ", " "}") like "[1 2 3 4]"
+	register.Grow(1 + 2*len(wd.register))
+	register.WriteRune('[')
+	for i, r := range wd.register {
+		if i != 0 {
+			register.WriteRune(' ')
+		}
+		register.WriteString(strconv.Itoa(r))
+	}
+	register.WriteRune(']')
```

Comment:
This chunk being repeated thrice, wouldn't it make sense to extract it in an intermediate function? I actually wonder if we don't already have this or something similar elsewhere in the code...

Reply:
Honestly, I don't like this part too :(

Since the codes(change `[]int` ->`string`) is more like a utility function and doesn't related with the `workdiff` logic here. Do we any place to put this kind of functions? I assume we need this function in some where else.